trulens/nn/attribution.py

```diff
@@ -245,101 +270,114 @@ class given by the first integer against the class given by the
+            batch_size = 1
+        else:
+            batch_size = model_inputs.first().shape[0]
 
-        # DoI supports tensor or list of tensor. unwrap args to perform DoI on
-        # top level list
+        param_msgs = [
+            f"float size = {B.floatX_size} ({B.floatX}); consider changing to a smaller type.",
+            f"batch size = {batch_size}; consider reducing the size of the batch you send to the attributions method."
+        ]
 
-        # Depending on the model_arg input, the data may be nested in data
-        # containers. We unwrap so that there operations are working on a single
-        # level of data container.
-        # TODO(piotrm): automatic handling of other common containers like the ones
-        # used for huggingface models.
-        # TODO(piotrm): this unwrapping may not be necessary any more
-        if isinstance(doi_val, DATA_CONTAINER_TYPE) and isinstance(
-                doi_val[0], DATA_CONTAINER_TYPE):
-            doi_val = doi_val[0]
+        doi_cut = self.doi.cut() if self.doi.cut() else InputCut()
 
-        if isinstance(doi_val, DATA_CONTAINER_TYPE) and len(doi_val) == 1:
-            doi_val = doi_val[0]
+        with memory_suggestions(*param_msgs):  # Handles out-of-memory messages.
+            doi_val: List[B.Tensor] = self.model._fprop(
+                model_inputs=model_inputs,
+                to_cut=doi_cut,
+                doi_cut=InputCut(),
+                attribution_cut=None,  # InputCut(),
+                intervention=model_inputs
+            )[0]
 
-        if accepts_model_inputs(self.doi):
-            D = self.doi(doi_val, model_inputs=model_inputs)
-        else:
-            D = self.doi(doi_val)
+        D = self.doi._wrap_public_call(doi_val, model_inputs=model_inputs)
 
-        n_doi = len(D)
+        n_doi = len(D[0])
+        doi_per_batch = self.doi_per_batch
+        if self.doi_per_batch is None:
+            doi_per_batch = n_doi
 
-        D = InternalInfluence.__concatenate_doi(D)
-        B = get_backend()
+        D = self.__concatenate_doi(D)
+        # Create a message for out-of-memory errors regarding doi_size.
+        # TODO: Generalize this message to doi other than LinearDoI:
+        doi_size_msg = f"distribution of interest size = {n_doi}; consider reducing intervention resolution."
+        doi_per_batch_msg = f"doi_per_batch = {doi_per_batch}; consider reducing this (default is same as the above)."
 
-        # Calculate the gradient of each of the points in the DoI.
-        qoi_grads = self.model.qoi_bprop(
-            qoi=self.qoi,
-            model_args=model_inputs.args,
-            model_kwargs=model_inputs.kwargs,
-            attribution_cut=self.slice.from_cut,
-            to_cut=self.slice.to_cut,
-            intervention=D,
-            doi_cut=doi_cut
-        )
+        # TODO: Consider doing the model_inputs tiling here instead of inside qoi_bprop.
+        effective_batch_size = doi_per_batch * batch_size
```

Comment:
I know this might because the review is early: is this used yet?
also if it is to be used later, thats fine. might not be bad to have separate PR